index.d.ts

```diff
@@ -0,0 +1,287 @@
+	/**
+	 * Change the spinner indent.
+	 */
+	indent: number;
+
+	/**
+	 * Start the spinner.
+	 *
+	 * @param text Set the current text.
+	 * @returns The spinner instance.
+	 */
+	start(text?: string): Ora;
+
+	/**
+	 * Stop and clear the spinner.
+	 *
+	 * @returns The spinner instance.
+	 */
+	stop(): Ora;
+
+	/**
+	 * Stop the spinner, change it to a green `✔` and persist the current text, or `text` if provided.
+	 *
+	 * @param text will persist text if provided
+	 * @returns The spinner instance.
+	 */
+	succeed(text?: string): Ora;
+
+	/**
+	 * Stop the spinner, change it to a red `✖` and persist the current text, or `text` if provided.
+	 *
+	 * @param text will persist text if provided
+	 * @returns The spinner instance.
+	 */
+	fail(text?: string): Ora;
+
+	/**
+	 * Stop the spinner, change it to a yellow `⚠` and persist the current text, or `text` if provided.
+	 *
+	 * @param text will persist text if provided
+	 * @returns The spinner instance.
+	 */
+	warn(text?: string): Ora;
+
+	/**
+	 * Stop the spinner, change it to a blue `ℹ` and persist the current text, or `text` if provided.
+	 *
+	 * @param text will persist text if provided
+	 * @returns The spinner instance.
+	 */
+	info(text?: string): Ora;
+
+	/**
+	 * Stop the spinner and change the symbol or text.
+	 *
+	 * @param options
+	 * @returns The spinner instance.
+	 */
+	stopAndPersist(options?: PersistOptions): Ora;
+
+	/**
+	 * Clear the spinner.
+	 * @returns The spinner instance.
+	 */
+	clear(): Ora;
+
+	/**
+	 * Manually render a new frame.
+	 * @returns The spinner instance.
+	 */
+	render(): Ora;
+
+	/**
+	 * Get a new frame.
+	 * @returns The spinner instance.
+	 */
+	frame(): Ora;
+}
+
+export interface Options {
+	/**
+	 * Text to display after the spinner.
+	 */
+	text?: string;
+
+	/**
+	 * Name of one of the provided spinners. See [`example.js`](https://github.com/BendingBender/ora/blob/master/example.js)
+	 * in this repo if you want to test out different spinners. On Windows, it will always use the line spinner as
+	 * the Windows command-line doesn't have proper Unicode support.
+	 *
+	 * Or an object like:
+	 *
+	 * @example
+	 *
+	 * {
+	 *     interval: 80, // optional
+	 *     frames: ['-', '+', '-']
+	 * }
+	 *
+	 * @default 'dots'
+	 */
+	spinner?: SpinnerName | Spinner;
+
+	/**
+	 * Color of the spinner.
+	 * @default 'cyan'
+	 */
+	color?: Color;
+
+	/**
+	 * Set to `false` to stop Ora from hiding the cursor.
+	 * @default true
+	 */
+	hideCursor?: boolean;
+
+	/**
+	 * Indent the spinner with the given number of spaces.
+	 * @default 0
+	 */
+	indent?: number;
+
+	/**
+	 * Interval between each frame.
+	 *
+	 * Spinners provide their own recommended interval, so you don't really need to specify this.
+	 * @default Provided by the spinner or 100
+	 */
+	interval?: number;
+
+	/**
+	 * Stream to write the output.
+	 *
+	 * You could for example set this to `process.stdout` instead.
+	 * @default process.stderr
+	 */
+	stream?: NodeJS.WritableStream;
```

Comment:
Import the stream interface instead of using the global.

Reply:
Not possible here, NodeJS is a global namespace, not a module you can import.

Reply:
It is possible: https://github.com/sindresorhus/hook-std/blob/f429639e9d60ae3259a17cd4aefaf62a9e4f15c1/index.d.ts#L1